Oqtane.Server/Repository/SiteRepository.cs

```diff
@@ -142,10 +142,13 @@ public Site GetSite(int siteId)
         }
 
         public void DeleteSite(int siteId)
-        {
-            Site site = db.Site.Find(siteId);
-            db.Site.Remove(site);
-            db.SaveChanges();
+        {       
+            if (db.Site.Count() > 1)
```

Comment:
Site.Count is not enough. Root(first) site have to be protected. Strange things happen when you do it.